### ISSUE TITLE
404 not found -- is it due to an expiring/revoked token?

### ISSUE THREAD

**Issue by dswhite42** [Caiyeon/goldfish#106]
It seems that after some time of running, Vault will eventually return nothing but the error "not found" when I visit its URL:

    {"time":"2017-07-03T14:39:23.667001984-05:00","remote_ip":"10.8.2.112","host":"myvault.example.com:8443","method":"GET","uri":"/","status":404, "latency":245,"latency_human":"245.214µs","bytes_in":0,"bytes_out":23}

Is this related to the wrap token expiring?  I had thought that this was fixed in Issue #27 , but even stopping and restarting Goldfish does not correct the issue.  Only after I walk through the whole installation procedure again do I get access.  (I do not know if **all** the following steps are required to re-initialize Goldfish - only that by doing all of this I can get Goldfish back up and running):

    pkill goldfish
    $VAULT policy-write goldfish vagrant/policies/goldfish.hcl
    $VAULT write auth/approle/role/goldfish role_name=goldfish secret_id_ttl=5m token_ttl=480h token_max_ttl=720h secret_id_num_uses=1 policies=default,goldfish
    $VAULT write auth/approle/role/goldfish/role-id role_id=goldfish
    $VAULT write -f transit/keys/goldfish
    $VAULT write secret/goldfish DefaultSecretPath="secret/" \
        TransitBackend="transit" \
        UserTransitKey="usertransit" \
        ServerTransitKey="goldfish" \
        BulletinPath="secret/bulletins/"
    export WRAP_TOKEN=`$VAULT write -f -wrap-ttl=20m -format=json auth/approle/role/goldfish/secret-id | $JQ -r .wrap_info.token`
    nohup $GOLDFISH \
    -goldfish_addr ":8443" \
    -vault_token $WRAP_TOKEN
    ... etc

Is it possible to modify either a Vault setting or a Goldfish setting in order to ensure that Goldfish is always up and running without user interaction?  (I'm willing to do something periodically by cron if need be, but it'd need to be something I could automate).  

Thanks!

**Comment by Caiyeon** [comment]
That's a 404 from goldfish webserver and not a 404 from vault. In which case it means your unzipped `public` folder is not in current working directory when you launch goldfish. 

In other words, the directory in which you launch goldfish using `nohup` must contain subdirectory `public` folder (with the html, js, and other stuff inside)

I would also highly recommend using v0.4.1 sometime, as it will be progressively more difficult to debug version specific issues

**Comment by dswhite42** [comment]
That is interesting, and a good bit of diagnosis - I had created a "start_goldfish.sh" startup script which did not chdir to $GOLDFISH_DIR before running goldfish.  I can correct that.  Would it make sense, though, to have Goldfish try to find public/ under its own binary's directory, rather than expect to find it under the working directory of where it was called from?

I am on 0.3.3, and will see if I can upgrade soon.

**Comment by Caiyeon** [comment]
I don't think the executable should be looking for a folder in its placed directory. This would mean the executable behaves differently based on where it's placed (i.e. usr/bin or local/bin).

However, you are not the first person to have been confused by this - I think you are the second or third? Clearly this is an issue. I'm closing this in favor of https://github.com/Caiyeon/goldfish/issues/107

Goldfish should fail to launch and report error if public folder is not found. This should at the very least notify the user of what is wrong.
